src/ObjWriting/ObjContainer/IWD/IWDWriter.cpp

```diff
@@ -0,0 +1,121 @@
+#include "IWDWriter.h"
+
+#include "../minizip/zip.h"
+
+#include <algorithm>
+#include <iostream>
+
+class IWDWriterImpl final : public IWDWriter
+{
+public:
+    explicit IWDWriterImpl(const std::string& iwdFileName, ISearchPath* assetSearchPath)
+        : m_filename(iwdFileName),
+          m_asset_search_path(assetSearchPath)
+    {
+        m_files.clear();
+        m_added_files.clear();

```

Comment:
Clearing vectors in constructors is not necessary because the vector will be initialize by the automatically called default constructor of `std::vector`